# Spotify next song

Use case: A producer at a major record label studio wants to make use of the data to ensure the next song makes a lot of moneeeeeeey. I have been assigned as the data analyst for this project. 

In [1]:
import numpy as np
import pandas as pd

In [31]:
topDF = pd.read_csv('top50.csv', encoding='iso-8859-1')
topDF.head()

Id                       Track.Name    Artist.Name           Genre  \
0   1                         Señorita   Shawn Mendes    canadian pop   
1   2                            China       Anuel AA  reggaeton flow   
2   3    boyfriend (with Social House)  Ariana Grande       dance pop   
3   4  Beautiful People (feat. Khalid)     Ed Sheeran             pop   
4   5      Goodbyes (Feat. Young Thug)    Post Malone         dfw rap   

   Beats.Per.Minute  Energy  Danceability  Loudness..dB..  Liveness  Valence.  \
0               117      55            76              -6         8        75   
1               105      81            79              -4         8        61   
2               190      80            40              -4        16        70   
3                93      65            64              -8         8        55   
4               150      65            58              -4        11        18   

   Length.  Acousticness..  Speechiness.  Popularity  
0      191               4             3          79  
1      302               8             9          92  
2      186              12            46          85  
3      198              12            19          86  
4      175              45             7          94

## How much BPM and length should the song have?

In [34]:
topDF[["Beats.Per.Minute", "Length."]].median()

Beats.Per.Minute    104.5
Length.             198.0
dtype: float64

Ans: Should have about 104 to 105 bpm and 3 minutes and 20 seconds

## Which singer should be signed up this time?

In [48]:
number_of_songs_by_singerDF = topDf[["Artist.Name", "Track.Name"]].groupby("Artist.Name").count()
number_of_songs_by_singerDF = number_of_songs_by_singerDF.sort_values("Track.Name", ascending=False)
number_of_songs_by_singerDF = number_of_songs_by_singerDF.rename(columns={"Track.Name":"Number of songs released"})
number_of_songs_by_singerDF.head(10)

Number of songs released
Artist.Name                               
Ed Sheeran                               4
Lil Nas X                                2
Shawn Mendes                             2
Marshmello                               2
Post Malone                              2
Sech                                     2
J Balvin                                 2
The Chainsmokers                         2
Billie Eilish                            2
Ariana Grande                            2